From 0593c0213faa8cc720eb9e364f80047fe24f746f Mon Sep 17 00:00:00 2001
From: Charles Karney <karney@users.sourceforge.net>
Date: Tue, 24 Sep 2019 10:23:08 -0400
Subject: [PATCH] Fix build with gcc-11

* backport only the compatibility with c++17 (-std=gnu++17 is default with gcc-11):
  https://sourceforge.net/p/geographiclib/discussion/1026621/thread/962b6bd5/
  from this huge commit: "Version 1.50 (2019-09-24)"

Upstream-Status: Pending
Signed-off-by: Martin Jansa <martin.jansa@lge.com>

---
 include/GeographicLib/Geoid.hpp               |  16 +-

diff --git a/include/GeographicLib/Geoid.hpp b/include/GeographicLib/Geoid.hpp
index 75723675..74ecaaa7 100644
--- a/include/GeographicLib/Geoid.hpp
+++ b/include/GeographicLib/Geoid.hpp
@@ -120,11 +120,7 @@ namespace GeographicLib {
     mutable real _v00, _v01, _v10, _v11;
     mutable real _t[nterms_];
     void filepos(int ix, int iy) const {
-      _file.seekg(
-#if !(defined(__GNUC__) && __GNUC__ < 4)
-                  // g++ 3.x doesn't know about the cast to streamoff.
-                  std::ios::streamoff
-#endif
+      _file.seekg(std::streamoff
                   (_datastart +
                    pixel_size_ * (unsigned(iy)*_swidth + unsigned(ix))));
     }
